src/navigator/offlineNavigator.js

```diff
@@ -110,8 +111,24 @@ export default function offlineNavigatorFactory(itemStore, responseStore) {
                 offlineJumpTableHelper[navigationActionName](params)
                     .then(function() {
                         lastJump = offlineJumpTableHelper.getLastJump();
-
-                        resolve(testContextBuilder.buildTestContextFromJump(testContext, testMap, lastJump));
+                        // new textContext doesn't know about item attempt
+                        // attempt is stored in itemStore
+                        // 1. get attempt from itemStore and increase it
+                        // 2. set it in new textContext
+                        // 3. store new attempt in itemStore
+                        // 4. return new textContext with right attempt
+                        Promise.all([
+                            testContextBuilder.buildTestContextFromJump(testContext, testMap, lastJump),
+                            itemStore.get(lastJump.item)
+                        ])
+                            .then(res => {
+                                const newTestContext = res[0];
+                                const itemFromStore = res[1];
+                                const newAttempt = itemFromStore.attempt ? itemFromStore.attempt + 1 : 1;
+                                newTestContext.attempt = newAttempt;
```

Comment:
Shouldn't it be performed by the testContextBuilder instead? The remaining number is already computed here: https://github.com/oat-sa/tao-test-runner-qti-fe/blob/17475a8a8f9ee721d5e23278f3dd9a08331588c9/src/helpers/testContextBuilder.js#L97
So, it will be better to also increase the attempt at the same time.
And then, you might update the item with the attempt value provided by the context

Reply:
fixed